packages/angular/build/src/tools/angular/transformers/lazy-routes-transformer.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import assert from 'node:assert';
+import { relative } from 'node:path/posix';
+import ts from 'typescript';
+
+export function lazyRoutesTransformer(
+  compilerOptions: ts.CompilerOptions,
+  compilerHost: ts.CompilerHost,
+): ts.TransformerFactory<ts.SourceFile> {
+  const moduleResolutionCache = compilerHost.getModuleResolutionCache?.();
+  assert(
+    typeof compilerOptions.basePath === 'string',
+    'compilerOptions.basePath should be a string.',
+  );
+  const basePath = compilerOptions.basePath;
+
+  return (context: ts.TransformationContext) => {
+    const factory = context.factory;
+
+    const visitor = (node: ts.Node): ts.Node => {
+      if (!ts.isObjectLiteralExpression(node)) {
+        return ts.visitEachChild(node, visitor, context);
+      }
+
+      let hasPathProperty = false;
+      let loadComponentOrChildrenProperty: ts.PropertyAssignment | undefined;
+      for (const prop of node.properties) {
+        if (!ts.isPropertyAssignment(prop) || !ts.isIdentifier(prop.name)) {
+          continue;
+        }
+
+        const propertyNameText = prop.name.text;
+        if (propertyNameText === 'path') {
+          hasPathProperty = true;
+        } else if (propertyNameText === 'loadComponent' || propertyNameText === 'loadChildren') {
+          loadComponentOrChildrenProperty = prop;
+        }
+
+        if (hasPathProperty && loadComponentOrChildrenProperty) {
+          break;
+        }
+      }
+
+      const initializer = loadComponentOrChildrenProperty?.initializer;
+      if (
+        !hasPathProperty ||
+        !initializer ||
+        !ts.isArrowFunction(initializer) ||
+        !ts.isCallExpression(initializer.body) ||
+        !ts.isPropertyAccessExpression(initializer.body.expression) ||
+        initializer.body.expression.name.text !== 'then' ||
+        !ts.isCallExpression(initializer.body.expression.expression) ||
+        initializer.body.expression.expression.expression.kind !== ts.SyntaxKind.ImportKeyword
+      ) {
+        return ts.visitEachChild(node, visitor, context);
+      }
```

Comment:
With default exports you still have a `then`.

```ts
() => import('./lazy/lazy.module').then(m => m.default)
```

Reply:
Angular Router automatically extracts the default export, so you can do `loadComponent: () => import('...')`, it will do the `.then(m => m.default)` for you.

Reply:
I wasn’t aware of this, I’ll update the transformer.